mz_zip.c

```diff
@@ -1,2771 +1,2794 @@
+}
+
+int32_t mz_zip_entry_is_symlink(void *handle) {
+    mz_zip *zip = (mz_zip *)handle;
+
+    if (zip == NULL)
+        return MZ_PARAM_ERROR;
+    if (zip->entry_scanned == 0)
+        return MZ_PARAM_ERROR;
+    if (mz_zip_attrib_is_symlink(zip->file_info.external_fa, zip->file_info.version_madeby) != MZ_OK)
+        return MZ_EXIST_ERROR;
+    if (zip->file_info.linkname == NULL || *zip->file_info.linkname == 0)
+        return MZ_EXIST_ERROR;
+
+    return MZ_OK;
+}
+
+int32_t mz_zip_entry_get_info(void *handle, mz_zip_file **file_info) {
+    mz_zip *zip = (mz_zip *)handle;
+
+    if (zip == NULL)
+        return MZ_PARAM_ERROR;
+
+    if ((zip->open_mode & MZ_OPEN_MODE_WRITE) == 0) {
+        if (!zip->entry_scanned)
+            return MZ_PARAM_ERROR;
+    }
+
+    *file_info = &zip->file_info;
+    return MZ_OK;
+}
+
+int32_t mz_zip_entry_get_local_info(void *handle, mz_zip_file **local_file_info) {
+    mz_zip *zip = (mz_zip *)handle;
+    if (zip == NULL || mz_zip_entry_is_open(handle) != MZ_OK)
+        return MZ_PARAM_ERROR;
+    *local_file_info = &zip->local_file_info;
+    return MZ_OK;
+}
+
+int32_t mz_zip_entry_set_extrafield(void *handle, const uint8_t *extrafield, uint16_t extrafield_size) {
+    mz_zip *zip = (mz_zip *)handle;
+
+    if (zip == NULL || mz_zip_entry_is_open(handle) != MZ_OK)
+        return MZ_PARAM_ERROR;
+
+    zip->file_info.extrafield = extrafield;
+    zip->file_info.extrafield_size = extrafield_size;
+    return MZ_OK;
+}
+
+static int32_t mz_zip_goto_next_entry_int(void *handle) {
+    mz_zip *zip = (mz_zip *)handle;
+    int32_t err = MZ_OK;
+
+    if (zip == NULL)
+        return MZ_PARAM_ERROR;
+
+    zip->entry_scanned = 0;
+
+    mz_stream_set_prop_int64(zip->cd_stream, MZ_STREAM_PROP_DISK_NUMBER, -1);
+
+    err = mz_stream_seek(zip->cd_stream, zip->cd_current_pos, MZ_SEEK_SET);
+    if (err == MZ_OK)
+        err = mz_zip_entry_read_header(zip->cd_stream, 0, &zip->file_info, zip->file_info_stream);
+    if (err == MZ_OK)
+        zip->entry_scanned = 1;
+    return err;
+}
+
+int64_t mz_zip_get_entry(void *handle) {
+    mz_zip *zip = (mz_zip *)handle;
+
+    if (zip == NULL)
+        return MZ_PARAM_ERROR;
+
+    return zip->cd_current_pos;
+}
+
+int32_t mz_zip_goto_entry(void *handle, int64_t cd_pos) {
+    mz_zip *zip = (mz_zip *)handle;
+
+    if (zip == NULL)
+        return MZ_PARAM_ERROR;
+
+    if (cd_pos < zip->cd_start_pos || cd_pos > zip->cd_start_pos + zip->cd_size)
+        return MZ_PARAM_ERROR;
+
+    zip->cd_current_pos = cd_pos;
+
+    return mz_zip_goto_next_entry_int(handle);
+}
+
+int32_t mz_zip_goto_file_entry(void *handle, int64_t cd_pos, const mz_zip_file *file_info) {
+    mz_zip *zip = (mz_zip *)handle;
+ 
+    if (zip == NULL || file_info == NULL)
+        return MZ_PARAM_ERROR;
+
+    if (cd_pos < zip->cd_start_pos || cd_pos > zip->cd_start_pos + zip->cd_size)
+        return MZ_PARAM_ERROR;
+
+    zip->cd_current_pos = cd_pos;
+    zip->file_info = *file_info;

```

Comment:
I think I would be opposed to being able to set the `file_info` from external source.